File: apps/render-ios-playground/render-ios-playground/Application/UseCases/RenderComponentUseCase.swift
import UIKit

/// Use case for rendering a schema into UIView
class RenderComponentUseCase {
    private let renderingService: ComponentRenderingService
    
    init(renderingService: ComponentRenderingService) {
        self.renderingService = renderingService
    }
    
    func execute(scenario: Scenario) async -> UIView? {
        return await MainActor.run {
            return renderingService.render(scenario.mainComponent)
        }
    }
    
    func execute(component: Component) -> UIView? {
        return renderingService.render(component)
    }
}

File: apps/render-android-playground/app/src/main/java/com/max_mrtnv/render_android_playground/application/usecases/RenderComponentUseCase.kt
package com.max_mrtnv.render_android_playground.application.usecases

import androidx.compose.runtime.Composable
import com.max_mrtnv.render_android_playground.domain.entities.Component
import com.max_mrtnv.render_android_playground.domain.entities.Scenario
import com.max_mrtnv.render_android_playground.domain.services.ComponentRenderingService

/**
 * Use case for rendering a scenario into Composables
 */
class RenderComponentUseCase(
    private val renderingService: ComponentRenderingService
) {
    @Composable
    fun execute(scenario: Scenario): Boolean {
        return renderingService.render(scenario.mainComponent)
    }
    
    @Composable
    fun execute(component: Component): Boolean {
        return renderingService.render(component)
    }
}
